pwiz_tools/Skyline/TestPerf/PerfCommandlineCreateImsDbTest.cs

```diff
@@ -43,55 +50,78 @@ public void CommandlineCreateImsDbPerfTest()
+                if (Equals(libName, badPath))
+                {
+                    imsdb = imsdb.Replace(TestFilesDirs[0].GetTestPath(string.Empty), badPath + "\\" + badPath);
+                }
                 var outCSV = TestFilesDirs[0].GetTestPath($"{root}.csv");
                 var output = RunCommand($"--in={template}",
                     $"--out={outSky}",
                     $"--import-transition-list={transitionList}",
                     $"--import-file={inFile}",
-                    $"--ionmobility-library-create={imsdb}",
+                    Equals(libName, badArgs) ? string.Empty : $"--ionmobility-library-create={imsdb}",
                     string.IsNullOrEmpty(libName) ? string.Empty : $"--ionmobility-library-name={libName}",
                     "--report-name=Precursor CCS",
                     $"--report-file={outCSV}");
                 if (Equals(libName, badName))
                 {
-                    // Expect failure because of illegal filename characters
-                    AssertEx.Contains(output, $"--ionmobility-library-create failed");
+                    // Expect failure because of illegal characters in imsdb filename
+                    AssertEx.Contains(output, 
+                        string.Format(Resources.ValueInvalidPathException_ValueInvalidPathException_The_value___0___is_not_valid_for_the_argument__1__failed_attempting_to_convert_it_to_a_full_file_path_,
+                            imsdb, "--ionmobility-library-create"));
+                    continue;
+                }
+                else if (Equals(libName, badPath))
+                {
+                    // Expect failure because of nonexistent subdirectories in path
+                    AssertEx.ContainsSimilar(output, Resources.CommandLine_SaveFile_Error__The_file_could_not_be_saved_to__0____Check_that_the_directory_exists_and_is_not_read_only_);
+                    continue;
+                }
+                else if (Equals(libName, badArgs))
+                {
+                    // Expect failure because of missing "--ionmobility-library-create"
+                    AssertEx.Contains(output, 
+                        string.Format(Resources.CommandArgs_WarnArgRequirment_Warning__Use_of_the_argument__0__requires_the_argument__1_,
+                            "--ionmobility-library-name", "--ionmobility-library-create"));
                     continue;
                 }
                 AssertEx.FileExists(imsdb);
-                AssertEx.Contains(output, $"Ion mobility library changed from \"None\" to \"{(string.IsNullOrEmpty(libName) ? root : libName)}\"");
 
-                // Compare to expected result
+                // Compare to expected result - may need to localize the expected copy to match the actual copy
                 var expected = File.ReadAllLines(TestFilesDirs[0].GetTestPath("ImsDbTest_expected.csv"));
```

Comment:
We have a lot of file comparison functions in AssertEx. This doesn't seem like a special case. So, either add it there or adapt what we have already.

Reply:
Good plan, will do